tests/Helpers/Factory.php

```diff
@@ -2,25 +2,39 @@
 
 namespace Algolia\AlgoliaSearch\Helper\Tests\Helpers;
 
+use Algolia\AlgoliaSearch\Helper\SearchClient;
+
 final class Factory
 {
+    private static $instance;
+
+    private $client;
+
     /**
-     * Returns a new index name based on the given `testName`.
+     * Returns a new index name based on the given `name`.
      *
-     * @param string $testName
+     * @param string $name
      *
      * @return string
      */
-    public static function getIndexName($testName)
+    public static function getIndexName($name)
     {
-        $phpversion = PHP_VERSION;
+        if (!self::$instance) {
+            self::$instance = getenv('TRAVIS') ? getenv('TRAVIS_JOB_NUMBER') : get_current_user();
+        }
 
-        date_default_timezone_set('UTC');
-
-        $date = date('Y-m-d_H:i:s');
-
-        $systemUsername = get_current_user();
+        return sprintf('php-helper_%s_%s_%s', date('Y-M-d_H:i:s'), self::$instance, $name);
+    }
 
-        return $phpversion.'_'.$date.'_'.$systemUsername.'_'.$testName;
+    /**
+     * Returns a new client.
+     *
+     * @param void
+     *
+     * @return \Algolia\AlgoliaSearch\Helper\SearchClient
+     */
+    public static function getClient()
+    {
+        return SearchClient::create(getenv('ALGOLIA_APP_ID'), getenv('ALGOLIA_API_KEY'));
```

Comment:
This should not be `ALGOLIA_APPLICATION_ID_1` and `ALGOLIA_ADMIN_KEY_1`?

Reply:
yes, but I had an issue with the API Key Dealer, @julienbourdeau is taking care of this, I tought it came from name of api key but in fact it's just because my prefix was not added.